packages/pglite-sync/src/index.ts

```diff
@@ -152,63 +219,143 @@ async function applyMessageToTable({
-    )
-    await pg.query(
-      `
-        UPDATE "${schema}"."${table}"
-        SET ${columns
-          .map((column, i) => '"' + column + '" = $' + (i + 1))
-          .join(', ')}
-        WHERE ${primaryKey
-          .map((column, i) => '"' + column + '" = $' + (columns.length + i + 1))
-          .join(' AND ')}
-      `,
-      [
-        ...columns.map((column) => data[column]),
-        ...primaryKey.map((column) => data[column]),
-      ],
-    )
-  } else if (message.headers.operation === 'delete') {
-    if (debug) {
-      console.log('deleting', data)
+
+    case 'delete': {
+      if (debug) console.log('deleting', data)
+      return await pg.query(
+        `
+            DELETE FROM "${schema}"."${table}"
+            WHERE ${primaryKey
+              .map((column, i) => '"' + column + '" = $' + (i + 1))
+              .join(' AND ')}
+          `,
+        [...primaryKey.map((column) => data[column])],
+      )
     }
-    await pg.query(
-      `
-        DELETE FROM "${schema}"."${table}"
-        WHERE ${primaryKey
-          .map((column, i) => '"' + column + '" = $' + (i + 1))
-          .join(' AND ')}
-      `,
-      [...primaryKey.map((column) => data[column])],
-    )
   }
 }
+
+interface GetShapeSubscriptionStateOptions {
+  pg: PGliteInterface | Transaction
+  shapeKey: ShapeKey
+}
+
+type ShapeSubscriptionState = Pick<ShapeStreamOptions, 'shapeId' | 'offset'>
+
+async function getShapeSubscriptionState({
+  pg,
+  shapeKey,
+}: GetShapeSubscriptionStateOptions): Promise<ShapeSubscriptionState | null> {
+  const result = await pg.query<{ shape_id: string; last_offset: string }>(
+    `
+    SELECT shape_id, last_offset
+    FROM ${subscriptionTableName}
+    WHERE shape_key = $1
+  `,
+    [shapeKey],
+  )
+
+  if (result.rows.length === 0) return null
+
+  const { shape_id: shapeId, last_offset: offset } = result.rows[0]
+  return {
+    shapeId,
+    offset: offset as Offset,
+  }
+}
+
+interface UpdateShapeSubscriptionStateOptions {
+  pg: PGliteInterface | Transaction
+  shapeKey: ShapeKey
+  shapeId: string
+  lastOffset: string
+}
+
+async function updateShapeSubscriptionState({
+  pg,
+  shapeKey,
+  shapeId,
+  lastOffset,
+}: UpdateShapeSubscriptionStateOptions) {
+  await pg.query(
+    `
+    INSERT INTO ${subscriptionTableName} (shape_key, shape_id, last_offset)
+    VALUES ($1, $2, $3)
+    ON CONFLICT(shape_key)
+    DO UPDATE SET
+      shape_id = EXCLUDED.shape_id,
+      last_offset = EXCLUDED.last_offset;
+  `,
+    [shapeKey, shapeId, lastOffset],
+  )
+}
+
+interface DeleteShapeSubscriptionStateOptions {
+  pg: PGliteInterface | Transaction
+  shapeKey: ShapeKey
+}
+
+async function deleteShapeSubscriptionState({
+  pg,
+  shapeKey,
+}: DeleteShapeSubscriptionStateOptions) {
+  await pg.query(`DELETE FROM ${subscriptionTableName} WHERE shape_key = $1`, [
+    shapeKey,
+  ])
+}
+
+const subscriptionTableName = `__electric_shape_subscriptions_metadata`
```

Comment:
I think we should create an `"electric"` schema, maybe with a name from a configuration option:

```ts
  pg = await PGlite.create({
    extensions: {
      electric: electricSync({
        metadataSchema: "myAlternativeSchema" // defaults to "electric"
      }),
    },
  })
```

Reply:
so it would be this my default:

`electric.shape_subscriptions_metadata`